include/deal.II/matrix_free/mapping_info.templates.h

```diff
@@ -121,331 +288,439 @@ namespace internal
           return 1;
         else return tria.begin()->diameter();
       }
-    }
 
 
 
-    template <int dim, typename Number>
-    void
-    MappingInfo<dim,Number>::initialize
-    (const dealii::Triangulation<dim>                         &tria,
-     const std::vector<std::pair<unsigned int,unsigned int> > &cells,
-     const std::vector<unsigned int>                          &active_fe_index,
-     const Mapping<dim>                                       &mapping,
-     const std::vector<dealii::hp::QCollection<1> >           &quad,
-     const UpdateFlags                                         update_flags_input)
-    {
-      clear();
-      const unsigned int n_quads = quad.size();
-      const unsigned int n_cells = cells.size();
-      const unsigned int vectorization_length =
-        VectorizedArray<Number>::n_array_elements;
-      Assert (n_cells%vectorization_length == 0, ExcInternalError());
-      const unsigned int n_macro_cells = n_cells/vectorization_length;
-      mapping_data_gen.resize (n_quads);
-      cell_type.resize (n_macro_cells);
+      template <int dim, typename Number>
+      struct CompressedCellData
+      {
+        CompressedCellData(const double expected_size)
+          :
+          data(FPArrayComparator<Number>(expected_size))
+        {}
+
+        std::map<Tensor<2,dim,Tensor<1,VectorizedArray<Number>::n_array_elements,Number> >
+        , unsigned int, FPArrayComparator<Number> > data;
```

Comment:
Yes, because I want to map a `Tensor` to an integer (storage location). Keep in mind that this part is simply moved here from another place, no new content.